test/browser/test_glfw3_hi_dpi_aware.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2023 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <GLFW/glfw3.h>
+#include <stdio.h>
+#include <assert.h>
+#include <string.h>
+#include <stdbool.h>
+#include <emscripten/html5.h>
+
+// installing mock devicePixelRatio (independent of the browser/screen resolution)
+static void installMockDevicePixelRatio() {
+  printf("installing mock devicePixelRatio...\n");
+  EM_ASM(
+      Browser.mockDevicePixelRatio = 1.0;
+      Browser.getDevicePixelRatio = () => { console.log("mock getDevicePixelRatio"); return Browser.mockDevicePixelRatio; };
+      );
+}
+
+static void setDevicePixelRatio(float ratio) {
+  printf("setDevicePixelRatio %.0f\n", ratio);
+  EM_ASM({
+    Browser.mockDevicePixelRatio = $0;
+    // mocking/simulating the fact that an event should be raised when devicePixelRatio changes
+    if (Browser.devicePixelRatioMQS) {
+      Browser.onDevicePixelRatioChange();
+    }
+  }, ratio);
+}
+
+static void setBrowserIsHiDPIAware(bool isHiDPIAware) {
+  printf("setBrowserIsHiDPIAware %s\n", isHiDPIAware ? "true" : "false");
+  EM_ASM({Browser.setHiDPIAware($0)}, isHiDPIAware ? 1 : 0);
+}
+
+static void checkWindowSize(GLFWwindow *window, int expectedWidth, int expectedHeight, float ratio) {
+  // first check the window size
+  int w, h;
+  glfwGetWindowSize(window, &w, &h);
+  assert(w == expectedWidth && h == expectedHeight);
+  
+  // second check the frame buffer size
+  glfwGetFramebufferSize(window, &w, &h);
+  assert(w == (int) (expectedWidth * ratio) && h == (int) (expectedHeight * ratio));
+}
+
+static bool getBrowserIsHiDPIAware() {
+  return EM_ASM_INT(return Browser.isHiDPIAware ? 1 : 0) != 0;
+}
+
+int main()
+{
+  
+    GLFWwindow *window;
```

Comment:
Is this function 2-space indented?  I can't tell easily from the github UI but it looks like it might be 3?

Reply:
Not sure what happened there. I remember seeing my editor complaining about not being able to process the clang format file. I wonder if that caused this issue.

In any case it is now fixed